brouter-core/src/main/java/btools/router/OsmNodeNamed.java

```diff
@@ -6,27 +6,89 @@
 package btools.router;
 
 import btools.mapaccess.OsmNode;
+import btools.util.CheapRulerSingleton;
 
 public class OsmNodeNamed extends OsmNode
 {
   public String name;
   public double radius; // radius of nogopoint
+  public double nogoWeight;  // weight for nogopoint
   public boolean isNogo = false;
-  
+
   @Override
   public String toString()
   {
-    return ilon + "," + ilat + "," + name;
+    if ( Double.isNaN(nogoWeight ) ) {
+      return ilon + "," + ilat + "," + name;
+    } else {
+      return ilon + "," + ilat + "," + name + "," + nogoWeight;
+    }
   }
-  
+
+  public double distanceWithinRadius(int lon1, int lat1, int lon2, int lat2, double totalSegmentLength) {
+    double[] lonlat2m = CheapRulerSingleton.getLonLatToMeterScales( (lat1 + lat2) >> 1 );
+
+    double realRadius = radius * 110984.;

```

Comment:
This should be updated depending whether https://github.com/abrensch/brouter/pull/130 gets merged before. Same below.